### PR TITLE
[WIP] Malware Subsystem

### PR DESCRIPTION
**Pull request by eeSPee**
## About The Pull Request

**Virus Disks**
New subsystem: Viruses. Cured with antiviruses. Installed through a disk on different consoles, and have various functions.

**Viruses**
Four viruses
* Honker - Makes console honk randomly
* Killer - Completely disables a console until treated
* Tider - Logs anyone using the console as Assistant McAssistantson.
* Keylogger - Copies the access of the last used ID on the console. Great for comms console/HoP console
* Cuban - Console explodes when used, damaging the user.

## Why It's Good For The Game

People wanted more stuff to do on the consoles.

## Progress
* [x] Implement Code
* [x] Ai Interactions
* [ ] Virus related Ai Powers
* [ ] Modular Console Viruses
* [ ] Standardized Consoles
* [ ] Testing
* [ ] Uplink kits
* [ ] Tech nodes/whatever
* [ ] Virus event

## Changelog
:cl:
add: viruses
/:cl:

### CODE REVIEW

**Comment by That0nePerson** [comment]
This looks great, but maybe use a different term for them? We've already got viro with viruses. Maybe something like worms or malware?

**Comment by VictorVow** [comment]
Malware sounds good

**Comment by eeSPee** [comment]
Renamed to malware. Thanks for pointing it out.

File: code/game/machinery/computer/virus.dm [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+	if (!suckers_id)
+		return
+	cached_id.access_txt = suckers_id.access_txt
+	cached_id.access = suckers_id.access
+	return cached_id
+
+
+//	---	SYNDICATE SELECTION	---
+
+/*	<3	LOGGER	<3	*
```
Comment (SmArtKar) [inline code/game/machinery/computer/virus.dm]: Forgot to close this little one!

**Comment by ike709** [comment]
Great idea, but seems rather barebones.

**Comment by francinum** [comment]
Likely best done after a rework of consoles towards a unified system. Current implementation seems to require heavy, intentional attaching to every form of computer.

**Comment by utilbird** [comment]
Why would we need malware content when we're playing on byond

File: code/game/machinery/computer/card.dm [hunk 2]
```diff
@@ -235,24 +238,27 @@ GLOBAL_VAR_INIT(time_last_changed_position, 0)
 		dat += " || Confirm Identity: "
 		var/S
 		var/list/paycheck_departments = list()
-		if(scan)
-			S = html_encode(scan.name)
+		var/obj/item/card/id/scanner_ID = scan
+		if (infection!=null)
```
Comment (utilbird) [inline code/game/machinery/computer/card.dm:242]: Why not just use `if(infection)`?

**Comment by eeSPee** [comment]
I'm reconsidering this as maybe a subsystem of a bigger console rework; basically, reworking pretty much every console as a modular console/computer or something like that. Not sure if that is a good idea, though.

 Should I keep this as is and try to standardize consoles a wee bit (ie ditch the whole thing)? Does it need more expanding? Should I discard it altogether?

**Comment by eeSPee** [comment]
I will rethink this.

**Pull request closed without merging**